internal/identity/gitlab.go

```diff
@@ -0,0 +1,127 @@
+package identity // import "github.com/pomerium/pomerium/internal/identity"
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	oidc "github.com/pomerium/go-oidc"
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/sessions"
+	"github.com/pomerium/pomerium/internal/version"
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultGitLabProviderURL = "https://gitlab.com"
+	revokeURL                = "https://gitlab.com/oauth/revoke"
+	oidcURL                  = "/oauth/userinfo"
```

Comment:
This can be parsed from the  `/.well-known/openid-configuration ` oidc endpoint see : https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig  and this comment https://github.com/desimone

Reply:
@desimone  This will require making a new request to get all the endpoints we want, that means we won't be hard-coding the endpoints at the expense of a request, right?

Reply:
That is right.